weather-app/src/components/Searchbar.vue

```diff
@@ -1,33 +1,261 @@
+<script>
+import axios from 'axios'
+let reAlpha = /^[a-z]+$/i
+
+export default {
+    
+    name: 'Searchbar',
+    data() {
+        return {
+            countries: null,
+            cities: null,
+            cityShow: true,
+            countryShow: true,
+            weatherShow: false,
+            fahrenheit: true,
+            // dom variables
+            cityName: null,
+            currentTemp: null,
+            maxTemp: null,
+            minTemp: null,
+            tempSymbol: null,
+            weatherImageUrl: null,
+            errorMessage: null,
+            country: null,
+            state: null
+            
+        }
+    },
+    
+    methods: {
+
+        async searchCountry(event, input) {
+            
+            // hide weather data and search city div
+            this.weatherShow = false;
+            this.cityShow = false; 
+            this.countryShow = true;
+
+            // only query database for alpa chars and backspace
+            if(reAlpha.test(event.key) || event.keyCode === 8) {
```

Comment:
In this instance, inverting this if check will help express the intent of your code more clearly, "Only search for countries if an alphabetical character or backspace was typed", by early-returning / exiting if that is not true. Consider:

```js
if (!reAlpha.test(event.key) && event.keyCode != 8) {
  return;
}

// Continue on doing API call here
```

Prefer fewer levels of indentation on the happy/success path when possible